src/proxy/hdrs/HdrHeap.cc

```diff
@@ -1168,12 +1178,9 @@ HdrStrHeap::free()
 char *
 HdrStrHeap::allocate(int nbytes)
 {
-  char *new_space;
-
-  if (m_free_size >= static_cast<unsigned>(nbytes)) {
-    new_space     = m_free_start;
-    m_free_start += nbytes;
-    m_free_size  -= nbytes;
+  if (_avail_size >= static_cast<unsigned>(nbytes)) {
+    char *new_space  = reinterpret_cast<char *>(this) + _total_size - _avail_size;
```

Comment:
I'd like to add a function that calculates this pointer with checking values.

Reply:
The checking is already present, on line 1181.  It results in a nullptr return if the checking fails.  The only place this function is called is in HdrHeap::allocate_str(), which does handle the null return value.